library/kafka_consumer_group.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+"""
+Ansible module for consumer group
+"""
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.pycompat24 import get_exception
+from kafka.errors import KafkaError
+
+from ansible.module_utils.kafka_lib_commons import (
+    module_commons, DOCUMENTATION_COMMON, get_manager_from_params,
+    maybe_clean_kafka_ssl_files
+)
+
+ANSIBLE_METADATA = {'metadata_version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: kafka_consumer_group
+short_description: Interact with kafka consumer groups
+description:
+     - Interact with kafka consumer groups.
+     - Not compatible with Kafka version < 0.11.0.
+author:
+    - Yassine MILHI
+options:
+  topics_partitions:
+    descritption:
+      - 'consumed topics partitions on which action will be performed'
+    required: True
+  consummer_group:
+    description:
+      - 'one consumer group name.'
+    required: True
+''' + DOCUMENTATION_COMMON
+
+EXAMPLES = '''
+    - name:  Delete offset for consumer group
+    kafka_consumer_group:
+        consummer_group: "{{ consummer_group | default('lolo_consumer')}}"
+        topics_partitions:
+        - 'lolo:0,1,2'
+        bootstrap_servers: "{{ ansible_ssh_host }}:9094"
+        api_version: "{{ kafka_api_version }}"
+        sasl_mechanism: "PLAIN"
+        security_protocol: "SASL_SSL"
+        sasl_plain_username: "admin"
+        sasl_plain_password: "{{ kafka_admin_password }}"
+        ssl_check_hostname: False
+        ssl_cafile: "{{ kafka_cacert | default('/etc/ssl/certs/cacert.crt') }}"
+'''
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            consummer_group=dict(type='str', required=True),
+            topics_partitions=dict(type='list', elements='str', required=True),
```

Comment:
Could you provide an API with topics which is mandatory and partitions which is optional? It's seems more simple to use if you just precise group and topics <=> You might need to fetch this group offsets.
It will also give you the ability to check parameters type topic==str and partition==int

Reply:
Done